src/NUnitFramework/tests/Attributes/TestCaseAttributeTests.cs

```diff
@@ -756,21 +756,22 @@ public void ExplicitTypeArgsWithoutParameters<T>()
             Assert.That(typeof(T), Is.EqualTo(typeof(int)));
         }
 
-        [TestCase(2, TypeArgs = new[] { typeof(int) })]
-        public void ExplicitTypeArgsWithUnrelatedParameters<T>(int input)
+        [TestCase("2", TypeArgs = new[] { typeof(long) })]
+        public void ExplicitTypeArgsWithUnrelatedParameters<T>(string input)
         {
-            Assert.That(typeof(T), Is.EqualTo(typeof(int)));
-            Assert.That(input, Is.EqualTo(2));
+            Assert.That(typeof(T), Is.EqualTo(typeof(long)));
+            Assert.That(input, Is.EqualTo("2"));
         }
 
-        [TestCase(2, TypeArgs = new[] { typeof(int) })]
-        public void ExplicitTypeArgsWithCompatibleParameters<T>(T input)
-        {
-            Assert.Pass();
-        }
+        [TestCase(2, TypeArgs = new[] { typeof(long) }, ExpectedResult = typeof(long))]
+        [TestCase(2L, TypeArgs = new[] { typeof(long) }, ExpectedResult = typeof(long))]
+        [TestCase(2, ExpectedResult = typeof(int))]
+        [TestCase(2L, ExpectedResult = typeof(long))]
+        public Type GenericMethodAndParameterWithExplicitOrImplicitTyping<T>(T input)
+            => typeof(T);
 
         [Test]
-        public void ExplicitTypeArgsWithIncompatibleParametersFailsAtRuntime()
+        public void ExplicitTypeArgsWithUnassignableParametersFailsAtRuntime()
```

Comment:
This test has the same issue as mentioned before. Don't run the test, the discovery turns the test into non-runnable.